tg_bot.py

```diff
@@ -250,16 +259,36 @@ def print_unsubscribe_button(msg):
         reply_markup=InlineKeyboardMarkup(custom_keyboard, one_time_keyboard=True))
 
 
+def print_subscription_status(update, msg, context): 
+    """
+    Prints current subscription status
+    """
+    user_id = str(update.effective_user.id)
+
+    # define subscription limit 
+    subscription_limit = scheduled_jobs[user_id].date() + datetime.timedelta(days=7)
+    # format date 
+    date_object = subscription_limit.strftime("%d-%m-%Y")
+
+    # check if exists a schedule job
+    if user_id in scheduled_jobs:
```

Comment:
Move this check upper, because you're accessing `scheduled_jobs[user_id]` before on line 269.

Or even better, just check it in the very beginning and return early

    if user_id not in scheduled_jobs:
        return

    # Continue with the rest

Reply:
done